internal/config/validation.go

```diff
@@ -331,35 +358,27 @@ func checkIfAWSInstanceTypeIsValid(userInput string) bool {
 		return false
 	}
 
-	// Now check if the user input is a supported family
-	// Note that we cannot directly use the family split from the Graviton check above, as some instances are directly specified by their full name and not just the family in general
-	for _, supportedFamily := range instancetypes.AWSSupportedInstanceFamilies {
-		supportedFamilyLowercase := strings.ToLower(supportedFamily)
-		if userDefinedFamily == supportedFamilyLowercase {
-			return true
+	if acceptNonCVM {
+		// Now check if the user input is a supported family
+		// Note that we cannot directly use the family split from the Graviton check above, as some instances are directly specified by their full name and not just the family in general
+		for _, supportedFamily := range instancetypes.AWSSupportedInstanceFamilies {
+			supportedFamilyLowercase := strings.ToLower(supportedFamily)
+			if userDefinedFamily == supportedFamilyLowercase {
+				return true
+			}
 		}
-	}
-
-	return false
-}
-
-// Validation translation functions for Azure & GCP instance type errors.
-func registerTranslateAzureInstanceTypeError(ut ut.Translator) error {
-	return ut.Add("azure_instance_type", "{0} must be one of {1}", true)
-}
-
-func (c *Config) translateAzureInstanceTypeError(ut ut.Translator, fe validator.FieldError) string {
-	// Suggest trusted launch VMs if confidential VMs have been specifically disabled
-	var t string
-
-	attestVariant := c.GetAttestationConfig().GetVariant()
-	if attestVariant.Equal(variant.AzureTrustedLaunch{}) {
-		t, _ = ut.T("azure_instance_type", fe.Field(), fmt.Sprintf("%v", instancetypes.AzureTrustedLaunchInstanceTypes))
 	} else {
-		t, _ = ut.T("azure_instance_type", fe.Field(), fmt.Sprintf("%v", instancetypes.AzureCVMInstanceTypes))
+		// Now check if the user input is a supported family
+		// Note that we cannot directly use the family split from the Graviton check above, as some instances are directly specified by their full name and not just the family in general
+		for _, supportedFamily := range instancetypes.AWSSNPSupportedInstanceFamilies {
+			supportedFamilyLowercase := strings.ToLower(supportedFamily)
+			if userDefinedFamily == supportedFamilyLowercase {
+				return true
+			}
+		}
 	}
```

Comment:
You can do something like
```go
	instances := instancetypes.AWSSNPSupportedInstanceFamilies
	if acceptNonCVM {
		instances = instancetypes.AWSSupportedInstanceFamilies
	}
```
instead of duplicating the whole for loop.
You can do something similar in the two touched translate funcs.

Reply:
:see_no_evil: